### PR TITLE
fix: support d&d of files without extension

### PR DESCRIPTION
**Pull request by dwelle**
fix https://github.com/excalidraw/excalidraw/issues/3158

### CODE REVIEW

File: src/data/blob.ts [hunk 1]
```diff
@@ -95,13 +94,7 @@ export const loadFromBlob = async (
         elements: clearElementsForExport(data.elements || []),
         appState: {
           theme: localAppState?.theme,
-          fileHandle:
-            blob.handle &&
-            ["application/json", MIME_TYPES.excalidraw].includes(
-              getMimeType(blob),
-            )
-              ? blob.handle
-              : null,
```
Comment (dwelle) [inline src/data/blob.ts:104]: I've removed this to allow for importing files without an extension. We're doing a check above to see if the file contains an appropriate JSON so we should be fine.

File: src/components/App.tsx [hunk 2]
```diff
@@ -3580,9 +3580,18 @@ class App extends React.Component<ExcalidrawProps, AppState> {
 
     const file = event.dataTransfer?.files[0];
     if (
-      file?.type === "application/json" ||
-      file?.name.endsWith(".excalidraw")
+      file?.type === MIME_TYPES.excalidrawlib ||
+      file?.name?.endsWith(".excalidrawlib")
```
Comment (tomayac) [inline src/components/App.tsx:3584]: We should probably move the allowed extensions into `constants.ts`.

**Review by tomayac — APPROVED** [review]
LGTM, with nits

File: src/components/App.tsx (hunk 2 above)
Comment (dwelle) [reply src/components/App.tsx:3630]: I'll do it in a separate PR coz I'm not sure what's the best way to do it.

**Pull request merged**